app/src/main/assets/native/ExoPlayerPlugin.js

```diff
@@ -78,308 +60,122 @@ export class ExoPlayerPlugin {
         }
     }
 
+    nextTrack() {}
+
+    previousTrack() {}
+
+    seek(ticks) {
+        this._nativePlayer.seek(ticks);
+    }
+
+    currentTime(ms) {
+        if (ms !== undefined) {
+            this._nativePlayer.seekMs(ms);
+        }
+        return this._currentTime;
+    }
+
+    duration(val) {
+        return null;
+    }
+
     /**
      * Get or set volume percentage as as string
      */
     volume(volume) {
         if (volume !== undefined) {
-            let volumeInt = parseInt(volume);
-            this._nativePlayer.setVolume(volumeInt);
+            this.setVolume(volume);
         }
         return null;
     }
 
-    setMute(mute) {
-        // Assume 30% as default when unmuting
-        this._nativePlayer.setVolume(mute ? 0 : 30);
+    getVolume() {}
+
+    setVolume(vol) {
+        let volume = parseInt(vol);
+        this._nativePlayer.setVolume(volume);
     }
 
+    volumeUp() {}
+
+    volumeDown() {}
+
     isMuted() {
         return false;
     }
 
-    seekable() {
-        return true;
+    setMute(mute) {
+        // Assume 30% as default when unmuting
+        this._nativePlayer.setVolume(mute ? 0 : 30);
     }
 
-    seek(ticks) {
-        this._nativePlayer.seek(ticks);
+    toggleMute() {}
+
+    paused() {
+        return this._paused;
     }
 
-    currentTime(ms) {
-        if (ms !== undefined) {
-            this._nativePlayer.seekMs(ms);
-        }
-        return this._currentTime;
+    pause() {
+        this._paused = true;
+        this._nativePlayer.pausePlayer();
     }
 
-    setSubtitleStreamIndex(index) {
-        this.subtitleStreamIndex = index;
+    unpause() {
+        this._paused = false;
+        this._nativePlayer.resumePlayer();
+    }
+
+    playPause() {
+        if (this._paused) {
+            this.unpause();
+        } else {
+            this.pause();
+        }
     }
 
     canSetAudioStreamIndex() {
         return false;
     }
 
-    setAudioStreamIndex(index) {
-        this.audioStreamIndex = index;
-    }
+    setAudioStreamIndex(index) {}
 
-    async changeSubtitleStream(index) {
-        var innerIndex = Number(index);
-        this.playbackManager.setSubtitleStreamIndex(innerIndex);
-        this.subtitleStreamIndex = innerIndex;
-    }
+    setSubtitleStreamIndex(index) {}
 
-    async changeAudioStream(index) {
-        var innerIndex = Number(index);
-        this.playbackManager.setAudioStreamIndex(innerIndex);
-        this.audioStreamIndex = innerIndex;
-    }
+    async changeAudioStream(index) {}
```

Comment:
Some previously implemented functions, like this one, are now empty. Why?

They can still be called when using remote control for example.

Reply:
Their previous implementation wasn't really handled either. So, they didn't work, and definitely won't in the new implementation. Still have to look into how to handle this with the SDK.

Reply:
Something for the fully native app then :p